typed_python/compiler/python_object_representation.py

```diff
@@ -175,6 +180,21 @@ def pythonObjectRepresentation(context, f):
     if f is abs:
         return TypedExpression(context, native_ast.nullExpr, AbsWrapper(), False)
 
+    if f is bool:
+        return TypedExpression(context, native_ast.nullExpr, BoolCastWrapper(), False)
```

Comment:
I don't think we want to have separate 'BytesCastWrapper', as this is the representation for 'bytes' itself as a free object. Before your change here, this would be 'PythonTypeObjectWrapper(f)' with 'f=bytes' (or whichever type). So you probably ought to go to the 'def convert_call(self, context, left, args, kwargs):' method of PythonTypeObjectWrapper and modify that code to dispatch to your convert_cast_X operations for these specific subclasses.

Alternatively, you could make subclasses of PythonTypeObjectWrapper called 'BytesTypeObjectWrapper', etc, and have the dispatch there.

Reply:
This has been done.  Makes things a lot simpler.